js/main.js

```diff
@@ -55,12 +57,65 @@ async function defineCardComponent() {
     //Binds the card's onclick events to flip and show the description popup.
     setupOnClickEvents() {
       this.back.onclick = (event) => this.flipCard();
-      this.front.onclick = (event) => alert("TODO: Description popup.");
+      this.front.onclick = (event) => this.showDetails();
+    }
+
+    showDetails() {
+      updateDetailsDialog(this.data, this.getImageURL());
+      DETAILS_DIALOG_A11Y.show();
```

Comment:
Sometimes I used the JS API (show), sometimes I added attribute tags to the HTML (hide) - if the HTML way didn't work I used JS and didn't dig too much into why the HTML way wasn't working :')